src/solver/conv_hip_implicit_gemm_wrw_v4r4_xdlops.cpp

```diff
@@ -294,41 +299,73 @@ PerformanceImplicitGemmWrwV4R4Xdlops::CalculateGridSize(const ConvolutionContext
     return std::make_tuple(GridSize, true);
 }
 
-int PerformanceImplicitGemmWrwV4R4Xdlops::CalculateGemmKBlocks(const ConvolutionContext& ctx) const
+std::tuple<int, int, int, int, int, bool>
+PerformanceImplicitGemmWrwV4R4Xdlops::CalculateGemmSizeAndGemmKBlock(
+    const ConvolutionContext& ctx) const
 {
```

Comment:
why put calculate GemmSize and GemmKBlock together?  GemmSizes are not tuning parameters, but GemmKBlock  calculate from GemmKPack, it is a tuning parameter.

Reply:
For this solver, GemmSizes becomes (dependent on) tuning paramemter: GemmG depends on GemmKBlock, and GemmKBlock depends on GemmMPerBlock and GemmNPerBlock.

Reply:
Gemmg doesn't need to be calculated in solver, does it? In solver, it represents group num and multiplies gemmkblock in kernel, which is actually a compromise.
In fact, I don't want to use for-loop to compute gemmmkblock multiple times. Of course, GemmSize is calculated at compile time, so it doesn't matter